src/utils/apiUtils.ts

```diff
@@ -23,11 +23,14 @@ export const fetchData = async <T>(
   }
 }
 
-export const fetchWidgetData = async (): Promise<
-  WidgetResponseBody | undefined
-> => {
+export const fetchWidgetData = async (
+  merchantId?: string,
+  productPrice?: number
+): Promise<WidgetResponseBody | undefined> => {
+  const requestBody = { merchantId: merchantId, price: productPrice }
```

Comment:
This can be simplified to `const requestBody = { merchantId, price: productPrice }` - can be simplified even further if you rename the parameter from productPrice to price